File: OsmAnd-java/src/main/java/net/osmand/binary/StringBundleWriter.java
package net.osmand.binary;

import net.osmand.binary.StringBundle.Item;

import java.util.Map.Entry;

public abstract class StringBundleWriter {

	private StringBundle bundle;

	public StringBundleWriter(StringBundle bundle) {
		this.bundle = bundle;
	}

	public StringBundle getBundle() {
		return bundle;
	}

	protected abstract void writeItem(String name, Item<?> item);

	public void writeBundle() {
		for (Entry<String, Item<?>> entry : bundle.getMap().entrySet()) {
			writeItem("osmand:" + entry.getKey(), entry.getValue());
		}
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/util/StringBundleWriter.kt
package net.osmand.shared.util

abstract class StringBundleWriter(private val bundle: StringBundle) {

	fun getBundle(): StringBundle {
		return bundle
	}

	protected abstract fun writeItem(name: String, item: StringBundle.Item<*>)

	open fun writeBundle() {
		for ((key, value) in bundle.getMap()) {
			writeItem("osmand:$key", value)
		}
	}
}
